modules/apps/object/object-rest-test/src/testIntegration/java/com/liferay/object/rest/internal/resource/v1_0/test/ObjectEntryResourceTest.java

```diff
@@ -5770,6 +5770,42 @@ public void testGetObjectEntriesFilteredBySystemDate() throws Exception {
 		}
 	}
 
+	@Test
+	public void testGetObjectEntriesPageActions() throws Exception {
+		JSONObject jsonObject1 = HTTPTestUtil.invokeToJSONObject(
+			null, _objectDefinition4.getRESTContextPath(), Http.Method.GET);
+
+		JSONObject actionsJSONObject1 = jsonObject1.getJSONObject("actions");
+
+		Assert.assertFalse(actionsJSONObject1.isNull("create"));
```

Comment:
If I understand the issue well, first you should test with a logged-in user and assert that `create`, `createBatch`, `deleteBatch` and  `updateBatch` are not null. Then you switch to an unauthenticated user (aka guest) and assert that actions list is null.

Reply:
Please check this [example](https://github.com/liferay/liferay-portal/blob/b4ac2f44f4b90f25e008bfa89bc68a2166e31696/modules/apps/headless/headless-builder/headless-builder-test/src/testIntegration/java/com/liferay/headless/builder/resource/test/HeadlessBuilderResourceTest.java#L2044).

Reply:
> and assert that create, createBatch, deleteBatch and updateBatch are not null.

You’re partially right. The logged-in admin user should only see the create and createBatch actions, and you should assert that the other three are not present. For the Guest or non-role user, no actions should be returned.

Reply:
I'm not sure if I understood your comment correctly, as the code isn't reflecting this, you continue to assert the five actions for both cases :confused: